Source/PeleCAmr.cpp

```diff
@@ -162,6 +172,32 @@ PeleCAmr::writePlotFileDoit(
       }
     }
 
+#ifdef PELEC_USE_SPRAY
+    if (!PeleC::spray_derive_vars.empty() && regular) {
+      int num_spray_derive = int(PeleC::spray_derive_vars.size());
+      PeleC::setupVirtualParticles(lev, finestLevel());
+      plotMFs[lev]->setVal(0., cnt, num_spray_derive);
+      // Compute derived spray variables for active particles
+      PeleC::theSprayPC()->computeDerivedVars(
+        *plotMFs[lev], lev, cnt, PeleC::spray_derive_vars,
+        PeleC::spray_fuel_names);
+      if (lev < finestLevel()) {
+        amrex::MultiFab tmp_plt(
+          boxArray(lev), DistributionMap(lev), num_spray_derive, 0,
+          amrex::MFInfo(), amr_level[lev]->Factory());
+        tmp_plt.setVal(0.);
+        // Compute derived spray variables for virtual particles under refined
+        // regions
+        PeleC::theVirtPC()->computeDerivedVars(
+          tmp_plt, lev, 0, PeleC::spray_derive_vars, PeleC::spray_fuel_names);
+        amrex::MultiFab::Add(
```

Comment:
Should this be an Add? Not a Copy?

Reply:
The active spray particles, which are in `theSprayPC`, only ever live on the finest level. The derived values are put into the plot MF from those first. Then the derived variables from the virtual particles, which are copies of active particles from finer levels to coarser levels, are added.

Reply:
ah gotcha!